extensions/ql-vscode/src/remote-queries/variant-analysis-manager.ts

```diff
@@ -176,6 +177,20 @@ export class VariantAnalysisManager extends DisposableObject implements VariantA
     return this.queue.pending;
   }
 
+  public getVariantAnalysisStorageLocation(variantAnalysisId: number): string {
+    return this.variantAnalysisResultsManager.getStorageDirectory(variantAnalysisId);
```

Comment:
Sorry I wasn't very clear with my comment in the other PR 🐱  

In my opinion, the responsibility and knowledge around where data is stored for a variant analysis, should live in the `VariantAnalysisManager` (not the `VariantAnalysisResultsManager`). If `VariantAnalysisResultsManager` needs it as well, it should be passed down to it in the relevant functions.

My reasoning for this is that we don't just store results, we store any information relevant to a variant analysis. Also, ideally the rest of the extension (e.g. query history) should interact with the variant analysis domain via the `VariantAnalysisManager` (which you've covered here! I'm just calling out how I think information should flow)

Reply:
So both the `VariantAnalysisManager` and `VariantAnalysisResultsManager` store the `storagePath` as a field? The results manager still needs access to that value, unless it retains a pointer back to the `VariantAnalysisManager`, and the circular dependency would seem like a bigger smell to me.

Ideally neither of these fields would be public, if we can expose methods like `getVariantAnalysisStorageLocation` that return the data that outside callers actually want.

If I've understood correctly then that approach sounds good to me.

Reply:
> So both the VariantAnalysisManager and VariantAnalysisResultsManager store the storagePath as a field?

Nope, just `VariantAnalysisManager` will have storagePath (and knowledge about how exactly to store variant analysis information). It will then pass that information to the `VariantAnalysisResultsManager` whenever it needs it. See https://github.com/github/vscode-codeql/pull/1631/commits/43de90f03d832ecae49906c7a1ca8d38970cad5e.

Does that make sense?

Reply:
Ok, I see what you mean now. That looks fine. 👍